e2e/libs/backupstore/base.py

```diff
@@ -0,0 +1,134 @@
+from abc import ABC, abstractmethod
+import time
+import hashlib
+from utility.utility import get_retry_count_and_interval
+
+class Base(ABC):
+
+    def is_backupTarget_s3(self, s):
+        return s.startswith("s3://")
+
+    def is_backupTarget_nfs(self, s):
+        return s.startswith("nfs://")
+
+    def backup_volume_path(self, volume_name):
+        volume_name_sha512 = \
+            hashlib.sha512(volume_name.encode('utf-8')).hexdigest()
+
+        volume_dir_level_1 = volume_name_sha512[0:2]
+        volume_dir_level_2 = volume_name_sha512[2:4]
+
+        backupstore_bv_path = "/backupstore/volumes/" + \
+            volume_dir_level_1 + "/" + \
+            volume_dir_level_2 + "/" + \
+            volume_name
+
+        return backupstore_bv_path
+
+    @abstractmethod
+    def get_backup_volume_prefix(self, client, volume_name):
+        return NotImplemented
+
+    def get_backup_target(self, client):
+        backup_target_setting = client.by_id_setting(SETTING_BACKUP_TARGET)
```

Comment:
`SETTING_BACKUP_TARGET` is not defined. Please check other places.

Reply:
Refined. Thank you!